logrotate/src/LogRotateHelper.ipp

```diff
@@ -403,7 +450,7 @@ bool LogRotateHelper::createCompressedFile(std::string file_name, std::string gz
         return false;
     }
 
-    int N;
+    size_t N;
     while ((N = fread(buffer, 1, buffer_size, input)) > 0) {
         gzwrite(output, buffer, N);
     }
```

Comment:
Maybe something like this?

```
    bool close_ok = (gzclose(output) == Z_OK);
    close_ok = (fclose(input) == 0)  && close_ok;

    return close_ok;
}
```

I will see if I can do a pull request to your pull request ;)